src/timesync/timesyncd-manager.c

```diff
@@ -1221,3 +1223,65 @@ static int manager_save_time_and_rearm(Manager *m, usec_t t) {
 
         return 0;
 }
+
+static const char* ntp_server_property_name[_SERVER_TYPE_MAX] = {
+        [SERVER_SYSTEM]   = "SystemNTPServers",
+        [SERVER_FALLBACK] = "FallbackNTPServers",
+        [SERVER_LINK] = "LinkNTPServers",
+        [SERVER_RUNTIME] = "RuntimeNTPServers",
+};
+
+static int ntp_server_emit_changed_strv(Manager *manager, char **properties) {
+        assert(manager);
+        assert(properties);
+
+        if (sd_bus_is_ready(manager->bus) <= 0)
+                return 0;
+
+        return sd_bus_emit_properties_changed_strv(
+                        manager->bus,
+                        "/org/freedesktop/timesync1",
+                        "org.freedesktop.timesync1.Manager",
+                        properties);
+}
+
+static int on_deferred_ntp_server(sd_event_source *s, void *userdata) {
+        Manager *m = ASSERT_PTR(userdata);
+        int r;
+        _cleanup_strv_free_ char **p = NULL;
+        m->deferred_ntp_server_event_source = sd_event_source_disable_unref(m->deferred_ntp_server_event_source);
+        for(int type = SERVER_SYSTEM; type < _SERVER_TYPE_MAX; type++)
+                if (m->ntp_server_change_mask & (1U << type) ) {
+                        if (strv_extend(&p, ntp_server_property_name[type]) < 0)
+                                log_oom();
+                        m->ntp_server_change_mask &= (~(1U << type));
```

Comment:
There's no need to remove bits from the mask one-by-one there. Just clear it once after the loop.